packages/playground/CoreApp/App.cpp

```diff
@@ -0,0 +1,75 @@
+#include "pch.h"
+#include "CoreApp.h"
+
+// This app demonstrates the usage of the Microsoft.ReactNative CoreApp APIs
+// use the preprocessor defines WITH_MODULES (true/false), FROM_JSON (true/false),
+// or neither to exercise the RNStartCoreAppWithModules, RNStartCoreApp, and RNStartCoreAppFromConfigJson APIs.
+#define DEFINE_MODULES
+
+// #define WITH_MODULES
+#define FROM_JSON
+
+#ifdef DEFINE_MODULES
+#include <ModuleRegistration.h>
+#include <NativeModules.h>
+#include <winrt/Microsoft.ReactNative.h>
+
+REACT_MODULE(LogBox)
+struct LogBox {
+  REACT_INIT(Initialize)
+  void Initialize(const winrt::Microsoft::ReactNative::ReactContext &) noexcept {
+    MessageBox(nullptr, L"Native Modules work!", L"Success", MB_OK);
+  }
+};
+
+struct ThisAppPackageProvider
+    : winrt::implements<ThisAppPackageProvider, winrt::Microsoft::ReactNative::IReactPackageProvider> {
+  void CreatePackage(winrt::Microsoft::ReactNative::IReactPackageBuilder const &packageBuilder) noexcept {
+    for (auto const *reg = winrt::Microsoft::ReactNative::ModuleRegistration::Head(); reg != nullptr;
+         reg = reg->Next()) {
+      packageBuilder.AddModule(reg->ModuleName(), reg->MakeModuleProvider());
+    }
+  }
+};
+
+extern "C" __declspec(dllexport) void *__cdecl MySpecialPackageProvider() {
+  auto provider = winrt::make<ThisAppPackageProvider>();
+  void *abi{nullptr};
+  winrt::copy_to_abi(provider, abi);
+  return abi;
+}
+
+#endif
+
+#ifdef FROM_JSON
+int __stdcall wWinMain(_In_ HINSTANCE, _In_opt_ HINSTANCE, _In_ PWSTR /*server*/, _In_ int /*showCommand*/) {
+  RNStartCoreAppFromConfigJson(L"app.config.json", nullptr);
```

Comment:
>nullptr [](http://example.com/codeflow?start=51&length=7)

What if FROM_JSON and WITH_MODULES are both defined? It seems that the code would not work in that case. Should it be changed to handle such case?

Reply:
This is just a sample so that folks know what kinds of things they can do but yeah we could